### PR TITLE
#253 Contributor stripe dashboard opens in a new tab.

### LINKED ISSUE #253: When navigating back from the "Stripe Dashboard" page, user should be redirected to the "Payout Methods" page
1. Go to the **Contributor Dashboard**
2. Go to **Payout Methods** page
3. Go to **Stripe Dashboard** page
4. Go back
Expected result: User should be taken back to step 2 **Payout Methods** page
image

Actual result: User is taken the the **Repository overview** page
image

**Suggestion**:  **Stripe Dashboard** page should be opened in a new tab

### PR DESCRIPTION
**Pull request by criske**
- also added one puzzle for wallet activation and one for contracts table optimization.

fixes #253

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by criske** [comment]
no bounty

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2020-12-22T13:16:52.246060``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: src/main/resources/public/js/wallets.js [hunk 1]
```diff
@@ -139,7 +139,8 @@ function getProjectWallets() {
  * @param owner Owner of the repo/project.
  * @param name Name of the repo.
  * @param type Type of the wallet (fake or stripe).
- * @todo #195#60min On frontend, enable wallet activation. Right now `activateWallet()` function is not used.
+ * @todo #253:30min On frontend, enable wallet activation.
+ *  Right now `activateWallet()` function is not used ('activate' button doesn't work).
```
Comment (amihaiemil) [inline src/main/resources/public/js/wallets.js:143]: @criske I don't think we need this TODO, wallet activation works fine since a month ago I think.

**Review by amihaiemil — COMMENTED** [review]
@criske just one question

File: src/main/resources/public/js/wallets.js (hunk 1 above)
Comment (criske) [reply src/main/resources/public/js/wallets.js:143]: @amihaiemil on my end is not working
Self XDSD

Comment (amihaiemil) [reply src/main/resources/public/js/wallets.js:143]: @criske the button is inactive because the Limit is 0. Try setting up a Limit, then it should work.

Comment (amihaiemil) [reply src/main/resources/public/js/wallets.js:143]: @criske and also at least a Payment Method.

Comment (criske) [reply src/main/resources/public/js/wallets.js:143]: thanks, it works now.

**Comment by charlesmike** [comment]
@amihaiemil Don't forget to close this ticket before the deadline (2020-12-22T13:16:52). You are past the first half of the allowed period.

**Comment by charlesmike** [comment]
@amihaiemil Looks like you've missed the task deadline (2020-12-22T13:16:52). You are now resigned from this task.

Please stop working on it, you will not be paid. I will assign it to someone else soon.

**Comment by charlesmike** [comment]
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``REV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.

**Comment by amihaiemil** [comment]
@criske thanks!

**Pull request merged**